File: backend/src/main/java/ma/zs/alc/service/facade/collaborator/quiz/ReponseCollaboratorService.java
package ma.zs.alc.service.facade.collaborator.quiz;

import java.util.List;
import ma.zs.alc.bean.core.quiz.Reponse;
import ma.zs.alc.dao.criteria.core.quiz.ReponseCriteria;
import ma.zs.alc.zynerator.service.IService;



public interface ReponseCollaboratorService extends  IService<Reponse,ReponseCriteria>  {

    List<Reponse> findByQuestionId(Long id);
    int deleteByQuestionId(Long id);
    long countByQuestionLibelle(String libelle);




}

File: frontend-bqckup/src/app/shared/service/collaborator/quiz/ReponseCollaborator.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';

import {environment} from '../../../../../environments/environment';

import {ReponseDto} from '../../../model/quiz/Reponse.model';
import {ReponseCriteria} from '../../../criteria/quiz/ReponseCriteria.model';
import {AbstractService} from "../../../../zynerator/service/AbstractService";


@Injectable({
    providedIn: 'root'
})
export class ReponseCollaboratorService extends AbstractService<ReponseDto, ReponseCriteria> {
    constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    get API() {
        return environment.apiUrlAlcservice + 'collaborator/reponse/';
    }

    public constrcutDto(): ReponseDto {
        return new ReponseDto();
    }

    public constrcutCriteria(): ReponseCriteria {
        return new ReponseCriteria();
    }
}
